.travis.yml

```diff
@@ -37,8 +38,6 @@ before_install:
     - git config --global user.name "TravisCI"
 
 install:
-    - sudo apt-get install python3-pip python3-nose
-    - git pull --recurse-submodules && git submodule update
```

Comment:
The PR makes sense to me. Thank you for the clean-up. 

Just one little thing, is there a specific reason why you want to remove the `git submodule update`?

Reply:
@adulau sorry about that, travis already runs the `git submodule update --init --recursive`, which provides the same support